giskard_vision/landmark_detection/detectors/base.py

```diff
@@ -0,0 +1,110 @@
+)
+
+
+class LandmarkDetectionBaseDetector(DetectorVisionBase):
+    """
+    Abstract class for Landmark Detection Detectors
+
+    Methods:
+        get_dataloaders(dataset: Any) -> Sequence[Any]:
+            Abstract method that returns a list of dataloaders corresponding to
+            slices or transformations
+
+        get_results(model: Any, dataset: Any) -> Sequence[ScanResult]:
+            Returns a list of ScanResult containing the evaluation results
+
+        get_scan_result(self, test_result) -> ScanResult:
+            Convert TestResult to ScanResult
+    """
+
+    @abstractmethod
+    def get_dataloaders(self, dataset: Any) -> Sequence[Any]: ...
+
+    def get_results(self, model: Any, dataset: Any) -> Sequence[ScanResult]:
+        dataloaders = self.get_dataloaders(dataset)
+
+        results = []
+        for dl in dataloaders:
+            test_result = TestDiff(metric=NMEMean, threshold=1).run(
+                model=model,
+                dataloader=dl,
+                dataloader_ref=dataset,
+            )
+
+            # Save example images from dataloader and dataset
+            current_path = str(Path())
+            os.makedirs(f"{current_path}/examples_images", exist_ok=True)
+            filename_examples = []
+
+            index_worst = 0 if test_result.indexes_examples is None else test_result.indexes_examples[0]
+
+            if isinstance(dl, FilteredDataLoader):
+                filename_example_dataloader_ref = f"{current_path}/examples_images/{dataset.name}_{index_worst}.png"
+                cv2.imwrite(
+                    filename_example_dataloader_ref, cv2.resize(dataset[index_worst][0][0], (0, 0), fx=0.3, fy=0.3)
+                )
+                filename_examples.append(filename_example_dataloader_ref)
+
+            filename_example_dataloader = f"{current_path}/examples_images/{dl.name}_{index_worst}.png"
+            cv2.imwrite(filename_example_dataloader, cv2.resize(dl[index_worst][0][0], (0, 0), fx=0.3, fy=0.3))
+            filename_examples.append(filename_example_dataloader)
+            results.append(self.get_scan_result(test_result, filename_examples, dl.name, len(dl)))
+
+        return results
+
+    def get_scan_result(self, test_result, filename_examples, name, size_data) -> ScanResult:
+        try:
+            from giskard.scanner.issues import IssueLevel
+        except (ImportError, ModuleNotFoundError) as e:
+            raise GiskardImportError(["giskard"]) from e
+
+        relative_delta = (test_result.metric_value_test - test_result.metric_value_ref) / test_result.metric_value_ref
+
+        if relative_delta > self.issue_level_threshold:
+            issue_level = IssueLevel.MAJOR
+        elif relative_delta > self.deviation_threshold:
+            issue_level = IssueLevel.MEDIUM
+        else:
+            issue_level = IssueLevel.MINOR
```

Comment:
that's not what I had in mind. Please recheck this: https://github.com/Giskard-AI/giskard-vision/pull/27#discussion_r1467902379 

the idea is not to show any issue in the scan report if it has `relative_delta < self.issue_level_threshold`
and reserve the `deviation_threshold` to define the level of the issue (between major and medium), we can fix the minor to be <5%.